Assets/Gothic-UnZENity-Core/Scripts/GameManager.cs

```diff
@@ -0,0 +1,185 @@
+using System;
+using GUZ.Core.Caches;
+using GUZ.Core.Context;
+using GUZ.Core.Manager;
+using GUZ.Core.Manager.Culling;
+using GUZ.Core.Manager.Settings;
+using GUZ.Core.World;
+using GUZ.Core;
+using GUZ.Core.Debugging;
+using GUZ.Core.Util;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace GUZ.Core
+{
+	[RequireComponent(typeof(TextureManager), typeof(FontManager))]
+	public class GameManager : MonoBehaviour, CoroutineManager, IGlobalDataProvider
+	{
+		public GameConfiguration config;
+		public GameObject xrInteractionManager;
+		public GameObject invalidInstallationPathMessage;
+
+		private VobMeshCullingManager _meshCullingManager;
+		private VobSoundCullingManager _soundCullingManager;
+		private BarrierManager _barrierManager;
+		private SkyManager _skyVisualManager;
+		private StationaryLightsManager _stationaryLightsManager;
+		private XRDeviceSimulatorManager _xrSimulatorManager;
+		private GameTime _gameTimeManager;
+		private MusicManager _gameMusicManager;
+		private GUZSceneManager _gameSceneManager;
+		private LoadingManager _gameLoadingManager;
+		private RoutineManager _npcRoutineManager;
+		private FileLoggingHandler _fileLoggingHandler;
+		private TextureManager _textureManager;
+		private FontManager _fontManager;
+
+		private GameSettings _settings;
+		private bool _isInitialised = false;
+
+		public GameSettings Settings => _settings;
+		public GameConfiguration Config => config;
+		public SkyManager Sky => _skyVisualManager;
+		public GameTime Time => _gameTimeManager;
+		public RoutineManager Routines => _npcRoutineManager;
+		public TextureManager Textures => _textureManager;
+		public GUZSceneManager Scene => _gameSceneManager;
+		public FontManager Font => _fontManager;
```

Comment:
That's funny. I didn't know, that we can define a field _private_ and as the appropriate Interface Field is marked as _public read_, it will override. It's kind of intransparent tbh. Could you please clone the public/private field values from the interface to these elements, please.